### PR TITLE
Cache Todos with pickle

### PR DESCRIPTION
**Pull request by untitaker**
This is an optimization I've been playing around. It scrapes 0.05s from the
execution speed in my case (with warm cache), so I guess it does something
right. The improvement may be bigger on different datasets.

I wonder where to store the pickle cache, storing it in vdir seems unsafe.

### CODE REVIEW

File: todoman/model.py [hunk 1]
```diff
@@ -236,30 +237,52 @@ class Database:
 
     def __init__(self, path):
         self.path = path
+        self._cache_path = os.path.join(self.path, '.todoman-cache')
```
Comment (hobarrera) [inline todoman/model.py]: $XDG_DATA_HOME sounds okay for this:

https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html#variables

**Comment by hobarrera** [comment]
> It scrapes 0.05s from the execution speed in my case (with warm cache)

I'm really not sure if the time difference warrant this. Do you have a very very small test vdir? I think that skipping non-VTODO entries by caching them as well might give a larger boots (since we might avoid opening dozens of irrelevant files).

**Comment by untitaker** [comment]
>I think that skipping non-VTODO entries by caching them as well might give a larger boost (since we might avoid opening dozens of irrelevant files).

I tried that one as well. It doesn't have any effect. The conditional that checks whether `\nBEGIN:VTODO` is contained in the file helps massively though.

**Comment by untitaker** [comment]
I fixed a bug where it didn't store entries for non-VTODO files.

**Comment by untitaker** [comment]
Ok now it scrapes 0.12s from `time todo list`

File: todoman/model.py (hunk 1 above)
Comment (untitaker) [inline todoman/model.py]: I opted for using `XDG_CACHE_HOME`

**Comment by hobarrera** [comment]
Nice. Skipping non-VTODO files is a massive difference.

**Comment by hobarrera** [comment]
LGTM, feel free to merge this.

**Pull request merged**